docs/modules/SDDs/pages/0024-naming-scheme-v2.adoc

```diff
@@ -0,0 +1,76 @@
+= SDD 0024 - Project Syn Tools Naming Scheme v2
+
+:sdd_author:    Tobias Brunner
+:sdd_owner:
+:sdd_reviewers:
+:sdd_date:      2020-10-23
+:sdd_status:    accepted
+include::partial$meta-info-table.adoc[]
+
+[NOTE]
+.Summary
+====
+Naming is hard, this page helps to make it easy. It describes the rules to name Project Syn tools.
+====
+
+== Motivation
+
+In xref:0002-naming-scheme.adoc[SDD 0002 - Naming Scheme] a naming scheme was defined for Project Syn tools which turned out to be sub-optimal.
+Having tools with non-descriptive names increases the (already relatively high) barrier of entry into the world of Project Syn.
+This SDD describes a more descriptive naming scheme which solves this issue.
+
+=== Goals
+
+* Naming scheme which is descriptive
+* Applicable to all new tools from now on
+
+=== Non-Goals
+
+* Renaming of existing tools (like Lieutenant, Steward and Commodore) as they will change over time anyways, ultimately being replaced by newer tools.
+
+== Design Proposal
+
+Tools must have descriptive display names of the form `Syn[ ${purpose}] ${type}`.
+By default the binary name for a tool would be `syn-$type[-$purpose]`, but deviations are allowed, see below.
+
+`$type`:: Describes the kind of the tool.
++
+Examples for types:
+
+* Operator
+* Agent
+* CLI
+
+`$purpose`:: Optional to describe purpose of the tool
+
+Examples:
+
+* syn-cli-compile
+* syn-operator-tenant
+* syn-agent
```

Comment:
These may need updating as well, depending on how we want binary names to work now. It may make sense to extend the examples with display names:

```suggestion
Examples:

* "Syn Compile CLI" -- `syn-cli-compile`
* "Syn Tenant Operator" -- `syn-operator-tenant`
* "Syn Agent" -- `syn-agent`
```